GateAdventures/Engine/Scenes/SceneManager.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System.Collections.Generic;
+using Microsoft.Xna.Framework;
+
+namespace GateAdventures.Engine.Scenes;
+
+/// <summary>
+/// Manage game scenes.
+/// </summary>
+public sealed class SceneManager
+{
+	private readonly List<IScene> _scenes;
+	private readonly GameServiceContainer _services;
+
+	/// <summary>
+	/// Get a list of active scenes.
+	/// </summary>
+	public IScene[] Scenes { get => _scenes.ToArray(); }
+
+	/// <summary>
+	/// Create a scene manager instance.
+	/// </summary>
+	/// <param name="services">The game services.</param>
+	public SceneManager(GameServiceContainer services)
+	{
+		_services = services;
+		_scenes = new List<IScene>();
+	}
+
+	/// <summary>
+	/// Stops all existing scenes and starts only the given scene.
+	/// </summary>
+	/// <param name="scene"></param>
+	public void Start(IScene scene)
+	{
+
+		for (int i = _scenes.Count - 1; i >= 0; i--)
+		{
+			Stop(_scenes[i]);
+		}
+
+		scene.Initialize(_services);
```

Comment:
One of the main issues I noticed when testing the branch was that the ball would reset to the center every time we changed scenes. We don't want to reinitialize on scene reload, that should only be called on the very first load. We only should load the content.

Reply:
The ball resets because it is a new instance of a scene which is by design. If we need persistent state then that state probably does not belong to the scene. We can consider creating a state manager in the future which is capable of saving to a file

Reply:
> The ball resets because it is a new instance of a scene which is by design

`Start()` shouldn't be a new instance. Imagine you're moving back and forth between a dungeon and the overworld, you don't want to be creating a new instance every single time. If we want a new reset instance, then we would call the constructor, and actually create a new instance, not use `Start()` as a pseudo constructor/reset.

> If we need persistent state then that state probably does not belong to the scene.

Even with a state manager, that would be a part of a scene, no? The NPCs, and objects are specific to a single scene, so that inherently means it belongs the scene. Maybe I'm misunderstanding what you mean here

> We can consider creating a state manager in the future which is capable of saving to a file

I'm not completely against this, it would be helpful to protect against crashes. Kind of like an autosave feature, however, the overhead of needing to read/write files is not insignificant, and if you did that every time you're loading and unloading a scene, if someone in the game is able to change scenes rapidly, that could lead to cases where you're reading the file before the previous close has overwritten it, which would then cause bad behavior.

Reply:
> Start() shouldn't be a new instance. Imagine you're moving back and forth between a dungeon and the overworld, you don't want to be creating a new instance every single time. If we want a new reset instance, then we would call the constructor, and actually create a new instance, not use Start() as a pseudo constructor/reset.

The reason I went with `Start` being passed a fresh instance of a scene and instances not being reused is so the following could be done:
```
SceneManager.Start(new TileScene("map-1"));
// And then later
SceneManager.Start(new TileScene("map-2")); // map-1 scene is stopped and disposed of automatically here.
```

> Even with a state manager, that would be a part of a scene, no? The NPCs, and objects are specific to a single scene, so that inherently means it belongs the scene. Maybe I'm misunderstanding what you mean here

I mean that the scene should not contain the business logic for saving the scene. You might need to call a `Save` method in a scene but that logic does not belong to the scene.

> I'm not completely against this, it would be helpful to protect against crashes. Kind of like an autosave feature, however, the overhead of needing to read/write files is not insignificant, and if you did that every time you're loading and unloading a scene, if someone in the game is able to change scenes rapidly, that could lead to cases where you're reading the file before the previous close has overwritten it, which would then cause bad behavior.

I think this detail should be discussed later but yes I agree straight to a file would not be ok. We would likely want to have a in memory segmented cache which is only persisted to a file under certain cases(Completing a level, opening the options menu,...). If memory becomes a problem it could be modeled as a queue where the last n number of scenes visited remain in memory.

Reply:
Agree with the last two points. For the first one, I'm undecided. I think I just need some time to think about it, so I won't block the PR on it. To me, stopping a scene does not necessarily mean completely delete the scene, just stop drawing and updating it. Allowing us to then restart the scene later, almost like a pause functionality